working/0678 - partial classes/feature-brainstorm.md

```diff
@@ -0,0 +1,199 @@
+# Partial classes
+
+Author: kevmoo@google.com
+
+Proposed solution to [partial classes (#252)](https://github.com/dart-lang/language/issues/252).
+Discussion about this proposal should go in [Issue #678](https://github.com/dart-lang/language/issues/678).
+
+## Motivation
+
+A critical use of code generation is adding boilerplate to hand-written classes.
+
+An ideal solution would provide:
+
+- Strict separation between human- and computer-generated code.
+- Allow computer-generated code to add members (functions, fields, properties,
```

Comment:
This does have some compile time implications - all codegen has to be done before a library can be summarized/outlined.

Reply:
Good point. Adding stubs now.